### ISSUE TITLE
Webcam video on Audio/Video testing page is very slow

### ISSUE THREAD

**Issue by yurivict** [qTox/qTox#3238]
Update speed is about 1 frame per second.
It is also hard to change any values on that page because GUI is slowed down too.

mplayer command on the same system plays video fine:

```
mplayer tv:// -tv driver=v4l2:width=1400:height=1050:fps=30:device=/dev/video0 -vo xv
```

Previous versions didn't suffer from this problem.

qTox-1.4.0 on FreeBSD-10.3

**Comment by zetok** [comment]
> Previous versions didn't suffer from this problem.

`1.3`, or `1.2.4` ?

Could you perhaps do https://github.com/tux3/qTox/wiki/Testing#regression-testing to see which commit introduced regression ?

**Comment by sudden6** [comment]
Can you also please upload the logfile?

**Comment by yurivict** [comment]
Resolutions 480p and less work fine, problem is only with the highest resolutions 720p and 1080p.

Version 1.3.0 works fine with resolution 1920x1080 30 fps.

Last working revision is 04ae894 (Mar 21), first broken revision is d28c371 (Mar 27).
Between them many revisions fail to build for me.

At some point it changes video format selection from NNNNxNNNN NNfps to NNNp. I think this is where regression is introduced.

Additionally, previous video format seems better because it is more explicit.

**Comment by zetok** [comment]
@yurivict could you do `git bisect` ?

**Comment by yurivict** [comment]
I can't use git disect because compilation fails for many intermediate revisions. But I am quite sure that it was introduced by the camera commit which killed fps display.

**Comment by zetok** [comment]
> I can't use git disect because compilation fails for many intermediate revisions.

Well, bisect is supposed to a degree handle the situation where compiling fails – have you tried `git bisect skip`? It should be used if e.g. compiling fails.

In any case, if commits from the particular PR seem to introduce the regression, could you perhaps try to revert that code & see whether it would work as expected?

**Comment by zetok** [comment]
Actually, reverting doesn't happen cleanly. Oh well, nvm.

Edit: Oh, and commit range if someone is interested: ce5cad4...3f89cd9

**Comment by yurivict** [comment]
Fixed in 1.5.1
